src/components/insignia.tsx

```diff
@@ -0,0 +1,91 @@
+"use client";
+import { useEffect, useState } from "react";
+import * as Models from "@/util/models";
+import Image from "next/image";
+
+const rarityColors = new Map<string, string>([
+  ["Common", "#FFFFFF"],
+  ["Uncommon", "#68D5ED"],
+  ["Rare", "#B36BFF"],
+  ["Unique", "#FF00FF"],
+  ["Epic", "#FFB400"],
+  ["Chronicle", "#FF6666"],
+  ["Legendary", "#FF7800"],
+  ["Mythic", "#FFB400"],
+]);
+
+export default function Insignia({ charId }: { charId: string }) {
+  const [insignia, setInsignia] = useState<Models.Insignia>();
```

Comment:
When making this a server side, no need to use useState nor useEffect. Just call and await the function in the service file.